### PR TITLE
Split gRPC and HTTP error utility into seperate packages

### PR DESCRIPTION
**Pull request by austinvazquez**
### Issue:
#4 

### Description:
By having a seperate packages, users can consume base package without pulling gRPC or HTTP as a dependency if not required.

This approach creates `errgrpc` and `errhttp`. The imports for users would be the following:

```go
import (
    "github.com/containerd/errdefs"
    "github.com/containerd/errdefs/errgrpc"
    "github.com/containerd/errdefs/errhttp"
)
```

### Other approaches considered:

Top level `grpc` and `http` packages. The imports for users would be the following:

```go
import (
    "github.com/containerd/errdefs"
    "github.com/containerd/errdefs/grpc"
    "github.com/containerd/errdefs/http"
)
```

My thought was this may clutter the top level directory, but this may be acceptable as the number of protocol packages would be finite.

### Testing:
Adds basic unit tests for there and back again error conversions for gRPC and HTTP error packages.

### CODE REVIEW

File: pkg/grpc/grpc.go [hunk 1]
```diff
@@ -14,7 +14,7 @@
    limitations under the License.
 */
 
-package errdefs
+package grpc
```
Comment (AkihiroSuda) [inline pkg/grpc/grpc.go:17]: errdefs.FromGRPC should be now called like grpc.ToNative ?

Comment (austinvazquez) [reply pkg/grpc/grpc.go:22]: Yeah, `grpc.FromGRPC` can be redundant. `grpc.ToNative` is nice naming imo. Pushed to try it out.

Comment (cpuguy83) [reply pkg/grpc/grpc.go:22]: Nit: maybe `grpcerror` or something?
I just dislike having package name conflicts that makes goimports trip up.

**Comment by dmcgowan** [comment]
Thanks for doing the split

> Top level grpc and http packages. The imports for users would be the following:

+1 to keeping it at the top. I see `pkg` more for packages that could live on their own or not directly extending the root package.

> Nit: maybe grpcerror or something?

+1 for keeping err (or error) in the package name (not the directory name).  Maybe to be like...

```
import (
    "github.com/containerd/errdefs"
    errgrpc "github.com/containerd/errdefs/grpc"
    errhttp "github.com/containerd/errdefs/http"
)
```
?

@cpuguy83 that what you were thinking?

**Comment by cpuguy83** [comment]
Not really, but that's fine.
For me I just end up fighting with goimports constantly with packages that have the same name and have to manually write the import statement.

**Comment by austinvazquez** [comment]
@cpuguy83, @dmcgowan thanks for the suggestions. Pushed changes to address. Landed with `errXXX` package naming to avoid conflicts with common packages like `http` and `grpc`. LMK your thoughts.

**Comment by dmcgowan** [comment]
Ok, works for me

**Review by cpuguy83 — APPROVED** [review]

**Review by dmcgowan — APPROVED** [review]

**Pull request merged**